lib/mongoid/core_ext/relations/options.rb

```diff
@@ -1,7 +1,21 @@
 module Mongoid
   module Relations
     module Options
-      COMMON << :versioned
+      VERSIONED_OPTIONS  = [:versioned].freeze
+
+      def validate!(options)
+        valid_options = options[:relation]::VALID_OPTIONS + COMMON + VERSIONED_OPTIONS
```

Comment:
I would like something like this here -> `options[:relation]::VALID_OPTIONS + common_options()`,
where
```ruby
def self.common_options
    super + [:versioned]
end
# or maybe have DSL for extending common options dedicated for extnsions
Mongoid::Relations::Options.add_common_option(:versioned)
```
, but all that requires changes in `mongoid` itself....